### PR TITLE
[feat]:[IAC-735]: Add Event consumer to IACM Manager

### PR DESCRIPTION
**Pull request by nassergonzalez**
## Describe your changes
Add Event consumer to IACM Manager

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51209)

### CODE REVIEW

**Comment by nassergonzalez** [comment]
trigger uts

**Comment by nassergonzalez** [comment]
trigger messagecheck

**Review by harshjain12 — APPROVED** [review]

**Review by r0610205 — APPROVED** [review]

**Comment by r0610205** [comment]
trigger uts

File: 310-iacm-manager/app/src/main/java/io/harness/app/IACMMAnagerConfigurationOverride.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import io.harness.ci.license.impl.CILicenseServiceImpl;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class IACMMAnagerConfigurationOverride {
```
[the remark this answers is not in the document]
Comment (r0610205) [reply 310-iacm-manager/app/src/main/java/io/harness/app/IACMMAnagerConfigurationOverride.java:23]: Well actually I see that you are applied overrides, but anyways unless you are using CI Service Module there is no need for it

File: 310-iacm-manager/app/src/main/java/io/harness/app/IACMManagerServiceModule.java [hunk 2]
```diff
@@ -320,4 +338,39 @@ public TelemetryConfiguration telemetryConfiguration() {
       }
     });
   }
+
+  private void registerEventListeners() {
+    final RedisConfig redisConfig = iacmManagerConfiguration.getEventsFrameworkConfiguration().getRedisConfig();
+    String orchestrationEvent = this.ciManagerConfigurationOverride.getOrchestrationEvent();
```
Comment (r0610205) [inline 310-iacm-manager/app/src/main/java/io/harness/app/IACMManagerServiceModule.java:344]: just plugin the values that you are using right here for orchestration event and serviceId

**Review by r0610205 — CHANGES_REQUESTED** [review]
Please, address the comments

File: 310-iacm-manager/app/src/main/java/io/harness/app/IACMMAnagerConfigurationOverride.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (nassergonzalez) [reply 310-iacm-manager/app/src/main/java/io/harness/app/IACMMAnagerConfigurationOverride.java:23]: Yep yep, I firstly copied everything so i could build the image and test in in QA but you are right, this class eventually is just used (at least for us) to have 2 strings so I will do that and remove this so i have less clutter

**Review by r0610205 — APPROVED** [review]

**Comment by nassergonzalez** [comment]
trigger uts

**Review by allofthesepeople — APPROVED** [review]

**Review by scottyw-harness — APPROVED** [review]

**Pull request merged**